diff --git a/configure.ac b/configure.ac
index c8b806f..d7a85c8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -89,8 +89,20 @@ AC_CHECK_HEADERS([sys/socket.h linux/types.h linux/irda.h], [irda_linux=yes], [i
 # endif
 ])
 
+AC_CHECK_HEADERS([irttp.h irphy.h], [irda_mac=yes], [irda_mac=no], [
+#if HAVE_IRTTP_H
+#   include <irttp.h>
+# endif
+#if HAVE_IRPHY_H
+#   include <irphy.h>
+# endif
+])
+
+AC_DEFINE([HAVE_IRDA], [1], [IrDA support])
+AM_CONDITIONAL([IRDA_MAC], [test "$irda_mac" = "yes"])
 AM_CONDITIONAL([IRDA], [test "$irda_win32" = "yes" || test "$irda_linux" = "yes"])
 
+
 # Checks for header files.
 AC_CHECK_HEADERS([linux/serial.h])
 AC_CHECK_HEADERS([IOKit/serial/ioss.h])
diff --git a/src/Makefile.am b/src/Makefile.am
index 39b5d5c..016c263 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -62,9 +62,14 @@ if OS_WIN32
 libdivecomputer_la_LIBADD += -lws2_32
 endif
 libdivecomputer_la_SOURCES += irda.h irda.c
+else
+if IRDA_MAC
+libdivecomputer_la_SOURCES += irda.h irda_mac.c
 else
 libdivecomputer_la_SOURCES += irda.h irda_dummy.c
 endif
+endif
+
 
 if OS_WIN32
 libdivecomputer_la_SOURCES += libdivecomputer.rc
